ui/e2e/swap.test.js

```diff
@@ -0,0 +1,286 @@
+require("@babel/polyfill");
+
+// configs
+const { MM_CONFIG, KEPLR_CONFIG } = require("./config.js");
+
+// extension
+const { metamaskPage } = require("./pages/MetaMaskPage");
+const { keplrPage } = require("./pages/KeplrPage.js");
+const { keplrNotificationPopup } = require("./pages/KeplrNotificationPopup.js");
+
+// services
+const { extractExtensionPackage } = require("./utils");
+const { useStack } = require("../test/stack");
+
+// utils
+const {
+  connectMetaMaskAccount,
+  connectKeplrAccount,
+  reconnectKeplrAccount,
+} = require("./helpers.js");
+
+// dex pages
+const { balancesPage } = require("./pages/BalancesPage.js");
+const { swapPage } = require("./pages/SwapPage.js");
+const { confirmSwapModal } = require("./pages/ConfirmSwapModal.js");
+const { connectPopup } = require("./pages/ConnectPopup.js");
+
+useStack("every-test");
+
+beforeAll(async () => {
+  // extract extension zips
+  await extractExtensionPackage(MM_CONFIG.id);
+  await extractExtensionPackage(KEPLR_CONFIG.id);
+
+  await metamaskPage.navigate();
+  await metamaskPage.setup();
+
+  await keplrPage.navigate();
+  await keplrPage.setup();
+
+  // goto dex page
+  await balancesPage.navigate();
+
+  // once keplr has finished setup, connection page will be invoked automatically
+  await context.waitForEvent("page");
+
+  await connectKeplrAccount();
+  await connectMetaMaskAccount();
+  await page.close();
+});
```

Comment:
Seems like there is a fair chunk of code here that looks like it is repeated. Would it make sense to have that in a setup file or as a shared function at all?

Reply:
Please check it out now. I extracted shared code into common setup file and helper functions.